core/dbt/parser/schemas.py

```diff
@@ -1005,6 +1056,132 @@ def _target_type(self) -> Type[UnparsedNodeUpdate]:
+                            msg = (
+                                f"Found {len(found_nodes)} matching disabled nodes for "
+                                f"{resource_type} '{target.name}'. Multiple nodes for the same "
+                                "unique id cannot be enabled in the schema file. They must be enabled "
+                                "in `dbt_project.yml` or in the sql files."
+                            )
+                            raise ParsingError(msg)
+                        versioned_model_node = self.manifest.disabled.pop(
+                            found_nodes[0].unique_id
+                        )[0]
+                        add_node_nofile_fn = self.manifest.add_disabled_nofile
+                else:
+                    versioned_model_node = self.manifest.nodes.pop(versioned_model_unique_id)
+                    add_node_nofile_fn = self.manifest.add_node_nofile
+
+                if versioned_model_node is None:
+                    warn_or_error(
+                        NoNodeForYamlKey(
+                            patch_name=versioned_model_name,
+                            yaml_key=target.yaml_key,
+                            file_path=source_file.path.original_file_path,
+                        )
+                    )
+                    continue
+
+                # update versioned node unique_id
+                versioned_model_node_unique_id_old = versioned_model_node.unique_id
+                versioned_model_node.unique_id = (
+                    f"model.{target.package_name}.{target.name}.{unparsed_version.formatted_v}"
+                )
+                # update source file.nodes with new unique_id
+                self.manifest.files[versioned_model_node.file_id].nodes.remove(
+                    versioned_model_node_unique_id_old
+                )
+                self.manifest.files[versioned_model_node.file_id].nodes.append(
+                    versioned_model_node.unique_id
+                )
+
+                # update versioned node fqn
+                versioned_model_node.fqn[-1] = target.name
+                versioned_model_node.fqn.append(unparsed_version.formatted_v)
+
+                # add versioned node back to nodes/disabled
+                add_node_nofile_fn(versioned_model_node)
+
+                # flatten columns based on include/exclude
+                version_refs: ParserRef = ParserRef.from_versioned_target(
+                    block.target, unparsed_version.v
+                )
+
+                versioned_model_patch = ParsedNodePatch(
+                    name=target.name,
+                    original_file_path=target.original_file_path,
+                    yaml_key=target.yaml_key,
+                    package_name=target.package_name,
+                    description=unparsed_version.description or target.description,
+                    columns=version_refs.column_info,
+                    meta={**target.meta, **unparsed_version.meta},
```

Comment:
This is where the problem with the meta dictionaries is happening. There's a method in core/dbt/context/context_config.py, "_add_config_call" which does the equivalent of "_merge_field_value" in core/dbt/contracts/graph/model_config.py.  Maybe you could pull out just the parts of _add_config_call which merges the dictionaries (BaseConfig.mergebejavior["update"]})

Reply:
taking a look, thank you for the pointer!

Reply:
Initially went down this route to merge "meta" nested within "config", but then found `deepmerge` in `dbt.utils` so went with that!

Reply:
Sounds good. That duplicative merge code is some technical debt that it would be nice to clean up sometime.